### ISSUE TITLE
tensor owner op incorrect

### ISSUE THREAD

**Issue by bhetherman** [flexflow/FlexFlow#60]
Tensor t = ff.conv2d(input, out_channels, 1, 1, 1, 1, 0, 0, AC_MODE_NONE); //owner is null
t = ff.batch_norm(t, false); //owner is Conv2d_11_100
t = ff.relu(t); //owner is Conv2d_11_100 but should be batch norm

Any idea why the owner ops are not propagating to the model layers correctly? It seems to be something with batch_norm because all the other layers are working?

**Comment by bhetherman** [comment]
Also i mean the owner ops of the input tensor

**Comment by eddy16112** [comment]
Are you sure you print the owner_op in the correct position of the code? Did you print t.owner_op after batch_norm? I can not reproduce the bug. Here is how the owner_op is set https://github.com/flexflow/FlexFlow/blob/master/src/runtime/model.cc#L151
I can not see why it works for others except batch_norm.

**Comment by bhetherman** [comment]
I added a print out where the best strategy is printed https://github.com/flexflow/FlexFlow/blob/3cb24aa54ad2ff98769671eda6c14b2eff1c71c2/src/runtime/model.cc#L1106

printf("inputs[");
    for (int i = 0; i < it->first->numInputs; i++){
      if(it->first->inputs[i].owner_op != NULL){
        if (i < it->first->numInputs - 1)
          printf("%s,", it->first->inputs[i].owner_op->name);
        else
          printf("%s", it->first->inputs[i].owner_op->name);
      }
    }
    printf("] ");

**Comment by bhetherman** [comment]
Im still seeing this issue.

If you change the best strategy print out to be:
```
printf("=========== Best Discovered Strategy ==========\n");
  std::map<Op*, ParallelConfig>::const_iterator it;
  for (it = best.begin(); it != best.end(); it++) {
    printf("[%s] num_dims(%d) dims[", it->first->name, it->second.nDims);
    for (int i = 0; i < it->second.nDims; i++)
      if (i < it->second.nDims - 1)
        printf("%d,", it->second.dim[i]);
      else
        printf("%d", it->second.dim[i]);
    printf("] device_ids[");
    for (int i = 0; i < it->second.num_parts(); i++)
      if (i < it->second.num_parts() - 1)
        printf("%d,", it->second.device_ids[i]);
      else
        printf("%d", it->second.device_ids[i]);
    printf("] ");
      
    printf("inputs[");
    for (int i = 0; i < it->first->numInputs; i++){
      if(it->first->inputs[i].owner_op != NULL){
        if (i < it->first->numInputs - 1)
          printf("%s,", it->first->inputs[i].owner_op->name);
        else
          printf("%s", it->first->inputs[i].owner_op->name);
      }
    }
    printf("] ");
    printf("\n");
  }
```

This is what i see:
=========== Best Discovered Strategy ==========
[Conv2D_77_100] num_dims(4) dims[1,1,1,1] device_ids[0] inputs[] 
[BatchNorm_101] num_dims(4) dims[1,1,1,1] device_ids[0] inputs[**Conv2D_77_100**] 
[ElementUnary_1_102] num_dims(4) dims[1,1,1,1] device_ids[0] inputs[**Conv2D_77_100**] 
...

When it should be:
=========== Best Discovered Strategy ==========
[Conv2D_77_100] num_dims(4) dims[1,1,1,1] device_ids[0] inputs[] 
[BatchNorm_101] num_dims(4) dims[1,1,1,1] device_ids[0] inputs[**Conv2D_77_100**] 
[ElementUnary_1_102] num_dims(4) dims[1,1,1,1] device_ids[0] inputs[**BatchNorm_101**] 
...

The model begins as 
  ```
// Add layers
  Tensor t = input;
  t = ff.conv2d(input, 64, 7, 7, 2, 2, 3, 3, AC_MODE_NONE);
  t = ff.batch_norm(t, false);
  t = ff.relu(t);
```

**Comment by jiazhihao** [comment]
@bhetherman The bug should have been fixed in commit 20752e12

**Comment by bhetherman** [comment]
This is fixed, Thanks @jiazhihao !
